mysql-test/suite/binlog_encryption/t/binlog_rotate_key.test

```diff
@@ -0,0 +1,154 @@
+
+--echo # master-bin.000004
+
+FLUSH LOGS;
+SELECT 5;
+SELECT 6;
+
+FLUSH LOGS;
+
+--echo # master-bin.000005
+
+SET SESSION debug="-d,check_consecutive_binlog_key_versions";
+
+# First check that each binlog is encrypted by trying to decrypt it using mysqlbinlog
+let datadir=`select @@datadir`;
+let filename= master-bin.000002;
+let local=$datadir/$filename;
+let outfile=$MYSQLTEST_VARDIR/tmp/binlog_master;
+--error 1
+exec $MYSQL_BINLOG $local --hexdump > $outfile;
+let filename= master-bin.000003;
+let local=$datadir/$filename;
+--error 1
+exec $MYSQL_BINLOG $local --hexdump >> $outfile;
+let filename= master-bin.000004;
+let local=$datadir/$filename;
+--error 1
+exec $MYSQL_BINLOG $local --hexdump >> $outfile;
+
+# Check that each generated binlog can be read by the server by using mysqlbinlog  with --read-from-remote-server
+# option
+let filename= master-bin.000002;
+let remote=--read-from-remote-server --protocol=tcp --host=127.0.0.1 --port=$MASTER_MYPORT -uroot $filename;
+let outfile=$MYSQLTEST_VARDIR/tmp/binlog_master-000001;
+exec $MYSQL_BINLOG --hexdump $remote >> $outfile;
+
+let filename= master-bin.000003;
+let remote=--read-from-remote-server --protocol=tcp --host=127.0.0.1 --port=$MASTER_MYPORT -uroot $filename;
+let outfile=$MYSQLTEST_VARDIR/tmp/binlog_master-000001;
+exec $MYSQL_BINLOG --hexdump $remote >> $outfile;
+
+let filename= master-bin.000004;
+let remote=--read-from-remote-server --protocol=tcp --host=127.0.0.1 --port=$MASTER_MYPORT -uroot $filename;
+let outfile=$MYSQLTEST_VARDIR/tmp/binlog_master-000001;
+exec $MYSQL_BINLOG --hexdump $remote >> $outfile;
+
+# Turning off binlog encryption to check if server can decrypt binlog with different percona_binlog keys versions
+# while not encrypting binary log
+--let $restart_parameters=restart:--log-bin=master-bin --encrypt-binlog=0
+--source include/restart_mysqld.inc
+
+--echo #master-bin.000006
+
+SELECT 7;
+SELECT 8;
+
+FLUSH LOGS;
+
+--echo #master-bin.000007
+
+# Check that current master-bin is not encrypted by reading it with mysqlbinlog
+let filename= master-bin.000006;
+let local=$datadir/$filename;
+exec $MYSQL_BINLOG $local --hexdump > $outfile;
+
+# Check that binlog encrypted with different percona_binlog key versions can be read by the server
+
+let remote=--read-from-remote-server --protocol=tcp --host=127.0.0.1 --port=$MASTER_MYPORT -uroot $filename;
+let outfile=$MYSQLTEST_VARDIR/tmp/binlog_master-000001;
+exec $MYSQL_BINLOG --hexdump $remote >> $outfile;
+
+let filename= master-bin.000003;
+let remote=--read-from-remote-server --protocol=tcp --host=127.0.0.1 --port=$MASTER_MYPORT -uroot $filename;
+let outfile=$MYSQLTEST_VARDIR/tmp/binlog_master-000001;
+exec $MYSQL_BINLOG --hexdump $remote >> $outfile;
+
+let filename= master-bin.000004;
+let remote=--read-from-remote-server --protocol=tcp --host=127.0.0.1 --port=$MASTER_MYPORT -uroot $filename;
+let outfile=$MYSQLTEST_VARDIR/tmp/binlog_master-000001;
+exec $MYSQL_BINLOG --hexdump $remote >> $outfile;
```

Comment:
If you keep the $outfile, then --remove_file $outfile

Reply:
Done